server/server.go

```diff
@@ -53,6 +54,8 @@ type Server struct {
 	routinesPool                  *safe.Pool
 	leadership                    *cluster.Leadership
 	defaultForwardingRoundTripper http.RoundTripper
+	metricsRegistry               metrics.Registry
+	metricsEnabled                bool
```

Comment:
`metricsRegistry` now always contains an implementation. When nothing is configured it will hold a `VoidRegistry`. This means we could write a util function like `metricsEnabled()` and it will return the `ok` of a type assertion on the metrics implementation. You would prefer this way?

Personally I am not sure about what is the more explicit option, but both is not hard to implement so you are free to choose :D